src/fastcs/backends/epics/ioc.py

```diff
@@ -40,95 +41,174 @@ def run(
                 always_update=True,
                 on_update=on_update,
             )
         case Int():
-            return builder.longOut(pv_name, always_update=True, on_update=on_update)
+            return builder.longOut(pv, always_update=True, on_update=on_update)
         case Float(prec):
-            return builder.aOut(
-                pv_name, always_update=True, on_update=on_update, PREC=prec
-            )
+            return builder.aOut(pv, always_update=True, on_update=on_update, PREC=prec)
         case String():
-            return builder.longStringOut(
-                pv_name, always_update=True, on_update=on_update
-            )
+            return builder.longStringOut(pv, always_update=True, on_update=on_update)
         case _:
             raise FastCSException(f"Unsupported type {type(datatype)}: {datatype}")
 
 
-def _create_and_link_command_pvs(mapping: Mapping) -> None:
+def _create_and_link_command_pvs(pv_prefix: str, mapping: Mapping) -> None:
     for single_mapping in mapping.get_controller_mappings():
         path = single_mapping.controller.path
         for attr_name, method in single_mapping.command_methods.items():
-            attr_name = attr_name.title().replace("_", "")
-            pv_name = f"{':'.join(path)}:{attr_name}" if path else attr_name
+            pv_name = attr_name.title().replace("_", "")
+            _pv_prefix = ":".join([pv_prefix] + path)
 
             _create_and_link_command_pv(
-                pv_name, MethodType(method.fn, single_mapping.controller)
+                _pv_prefix,
+                pv_name,
+                attr_name,
+                MethodType(method.fn, single_mapping.controller),
             )
 
 
-def _create_and_link_command_pv(pv_name: str, method: Callable) -> None:
+def _create_and_link_command_pv(
+    pv_prefix: str, pv_name: str, attr_name: str, method: Callable
+) -> None:
     async def wrapped_method(_: Any):
         await method()
 
-    builder.aOut(pv_name, initial_value=0, always_update=True, on_update=wrapped_method)
+    record = builder.aOut(
+        f"{pv_prefix}:{pv_name}",
+        initial_value=0,
+        always_update=True,
+        on_update=wrapped_method,
+    )
+
+    _add_pvi_info(record, pv_prefix, attr_name, "x")
+
+
+def _add_pvi_info(
+    record: RecordWrapper,
+    prefix: str,
+    name: str,
+    access_mode: Literal["r", "w", "rw", "x"],
+):
+    """Add an info tag to a record to include it in the PVI for the controller.
+
+    Args:
+        record: Record to add info tag to
+        prefix: PV prefix of controller
+        name: Name of parameter to add to PVI
+        access_mode: Access mode of parameter
+    """
+    record.add_info(
+        "Q:group",
+        {
+            f"{prefix}:PVI": {
+                f"pvi.{name}.{access_mode}": {
+                    "+channel": "NAME",
+                    "+type": "plain",
+                    "+trigger": f"pvi.{name}.{access_mode}",
```

Comment:
We would like to do this long term instead:
https://github.com/PandABlocks/PandABlocks-ioc/issues/120

Reply:
I think this is done now. I also added a system test to try introspecting the PVI structure - I can add some more specific checks here if you can suggest any.